pkg/container/vector/vector.go

```diff
@@ -21,46 +21,39 @@ import (
 	"unsafe"
 
 	"github.com/matrixorigin/matrixone/pkg/common/bitmap"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
 	"github.com/matrixorigin/matrixone/pkg/common/mpool"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
-	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 )
 
-// XXX Moved vector from types.go to vector.go
-// XXX Deleted vector interface, which was commented out and outdated anyway.
-/* Vector vector
- * origin true:
- * 				count || type || bitmap size || bitmap || vector
- * origin false:
- *  			count || vector
- */
+const (
+	FLAT     = iota // flat vector represent a uncompressed vector
+	CONSTANT        // const vector
+	DIST            // dictionary vector
+)
+
+// Vector represent a column
 type Vector struct {
-	// XXX There was Ref and Link, from the impl, it is totally wrong stuff.
-	// Removed.
-	Typ types.Type
-	Col interface{}  // column data, encoded Data
-	Nsp *nulls.Nulls // nulls list
+	// vector's class
+	class int
+	// type represent the type of column
+	typ types.Type
+	nsp *nulls.Nulls // nulls list
 
-	original bool
-	// data of fixed length element, in case of varlen, the Varlena
-	data []byte
 	// area for holding large strings.
 	area []byte
 
-	// some attributes for const vector (a vector with a lot of rows of a same const value)
-	isConst bool
-	length  int
+	length int
 
 	// tag for distinguish '0x00..' and 0x... and 0x... is binary
 	// TODO: check whether isBin should be changed into array/bitmap
 	// now we assumpt that it can only be true in the case of only one data in vector
 	isBin bool
-
-	// idx for low cardinality scenario.
-	idx any
+	// data of fixed length element, in case of varlen, the Varlena
+	col *reflect.SliceHeader
```

Comment:
why not just []byte?    that is we keep data.   get rid of this col.   always use unsafe to cast data into correct slice.